### ISSUE TITLE
Do not reference types.ClassType or types.InstanceType in jelly.py on Python 3

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#8646]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)| @rodrigc reported|
|-|-|
|Trac ID|trac#8646|
|Type|enhancement|
|Created|2016-07-21 13:13:51Z|
|Branch|https://github.com/twisted/twisted/tree/8646-rodrigc-noclasstype|

types.ClassType and types.InstanceType are in jelly.py and need
to still be there for Python 2 compatibility.
There are complteely gone from Python 3.
Special care must be taken to not reference these on Python 3,
but preserve functionality on Python 2.

<details><summary>Searchable metadata</summary>

```
trac-id__8646 8646
type__enhancement enhancement
reporter__rodrigc rodrigc
priority__normal normal
milestone__Python_3_x Python-3.x
branch__8646_rodrigc_noclasstype 8646-rodrigc-noclasstype
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__None None
time__1469106831230238 1469106831230238
changetime__1474198049187875 1474198049187875
version__None None
owner__GitHub__noreply_____ GitHub <noreply@...>

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)<a name="note_1"></a>|@rodrigc commented|
|-|-|

https://github.com/twisted/twisted/pull/381

**Comment by twisted-trac** [comment]
|<img alt="GitHub's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_2"></a>|GitHub set owner to GitHub<br>GitHub set status to `closed`|
|-|-|

In changeset 0538c2410148bf571dbf95ef125a15d42b685c08
```
#!CommitTicketReference repository="" revision="0538c2410148bf571dbf95ef125a15d42b685c08"
Merge pull request #381 from twisted/8646-rodrigc-noclasstype

Author: rodrigc
Reviewer: markrwilliams
Fixes: #8646

Use _OldStyleClass and _OldStyleInstance for ClassType and InstanceType
```

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_3"></a>|@hawkowl commented|
|-|-|

[mass edit] Removing review from closed tickets.
